### ISSUE TITLE
Synchronization reliability on Android platform

### ISSUE THREAD

**Issue by dyura** [standardnotes/forum#341]
Hello! 
It is regarding the synchronization in Android application after the device was off-line. "Notes" on Android is not using services. Not using services might be reasonable. However, without any service "Notes" cannot assure the synchronization after the device becomes again connected to the network. If we are intensively using an Android device, when being off-line, we have a good chance, that "Notes" will be killed by Android before synchronization can start. A service would be a correct solution, but avoiding too much extra work, I would suggest two different workarounds. First is very simple, just to set a reminder. When an update was done in offline mode, Notes can send notification, which will mean, that  the content is not synchronized yet. It will remind to start "Notes" manually when network will be back.
The second workaround could be  scheduled synchronization via "alarm manager".

Thanks,
Yury

**Comment by mobitar** [comment]
Hey @dyura, in this case, syncing would happen in two scenarios:

1. Immediately when you open the app.
2. If you are offline while the app is open, then come online while the app is open, it will not sync right away, but will sync within the next 30 second timer. 

But if you are offline then come online, then open Standard Notes, it should sync right away. Are you noticing other behavior?

**Comment by dyura** [comment]
Hi ,Mo,
“it will not sync right away, but will sync within the next 30 second timer.”  It is not always so.  I am offline   every day for about 1.5 hour, and I had an opportunity to observe. The statement “while the app is open” is ambiguous. If you mean, that after offline changes we will keep the application on the screen, that means not using other applications, then I agree. It is very unlikely that Android will kill the process. However, if after changes you will open another applications, then the picture is very different. Just reading offline messages might not affect Notes. But work with memory consuming applications, such as spreadsheets, word processors, pdf viewers, movie players, etc, will force Android to kill all inactive processes, and it is 100% that Notes will be killed while we are offline.  Then when we come online, we have to keep in mind to start Notes again. Currently, if my offline changes are important, I set reminders to check later, if Notes were synchronized. That is why  I  suggest adding notifications to Notes. 
Alarmmanager  could be also useful, but it requires more logic. 

Yury

**Comment by mobitar** [comment]
Oh I see. So you're saying that you might make important offline changes in the Android app, close the app, then, maybe later, you come on your Desktop app or something and expect to see your changes but don't? That's true, it wouldn't automatically sync your changes. We don't do any sort of background activity. But I can see why that can be useful. I'll keep this in mind. We use React Native, so however we build this needs to be cross-compatible with iOS, which makes this a little tricky out of the box.

**Comment by dyura** [comment]
Exactly, Mo! You got it. I had few moments
when I left my phone behind, and couldn't find my changes on Web.
I realize that adding background processes would be a bigger task, but
reminding via standard Android notifications could be more realistic and
helpful.  Of course, it is not a priority,  but Notes could push a
notification right after update, and cancel it after synch. When Notes will
be killed, the notification will stay.
I found a package for React Native:
com.dieam.reactnativepushnotification.ReactNativePushNotificationPackage;
I don’t know how well it works, also I don’t know about Android 8 as
NotificationCompat.Builder was changed for channel support. But, anyway it
is an option.
Thanks,
Yury

On 10/11/18, Mo Bitar <notifications@github.com> wrote:
> Oh I see. So you're saying that you might make important offline changes
in
> the Android app, close the app, then, maybe later, you come on your
Desktop
> app or something and expect to see your changes but don't? That's true, it
> wouldn't automatically sync your changes. We don't do any sort of
background
> activity. But I can see why that can be useful. I'll keep this in mind. We
> use React Native, so however we build this needs to be cross-compatible
with
